error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJStreamRules.java

```diff
@@ -0,0 +1,81 @@
+package tech.picnic.errorprone.refasterrules;
+
+import static com.google.errorprone.refaster.ImportPolicy.STATIC_IMPORT_ALWAYS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.errorprone.refaster.Refaster;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.UseImportPolicy;
+import java.util.Collection;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.assertj.core.api.AbstractAssert;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+
+@OnlineDocumentation
+final class AssertJStreamRules {
+  private AssertJStreamRules() {}
+
+  static final class AssertThatFilteredOn<T> {
+    @BeforeTemplate
+    AbstractAssert<?, ?> before(Stream<T> stream, Predicate<? super T> predicate) {
+      return assertThat(stream.filter(predicate));
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    AbstractAssert<?, ?> after(Stream<T> stream, Predicate<? super T> predicate) {
+      return assertThat(stream).filteredOn(predicate);
+    }
+  }
+
+  static final class AssertThatNoneMatch<T> {
+    @BeforeTemplate
+    void before(Stream<T> stream, Predicate<? super T> predicate) {
+      assertThat(stream).filteredOn(predicate).isEmpty();
+    }
+
+    @BeforeTemplate
+    void before2(Stream<T> stream, Predicate<? super T> predicate) {
+      Refaster.anyOf(
+          assertThat(stream.anyMatch(predicate)).isFalse(),
+          assertThat(stream.noneMatch(predicate)).isTrue());
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Stream<T> stream, Predicate<? super T> predicate) {
+      assertThat(stream).noneMatch(predicate);
+    }
+  }
+
+  static final class AssertThatAnyMatch<T> {
+    @BeforeTemplate
+    AbstractAssert<?, ?> before(Stream<T> stream, Predicate<? super T> predicate) {
+      return Refaster.anyOf(
+          assertThat(stream).filteredOn(predicate).isNotEmpty(),
+          assertThat(stream.anyMatch(predicate)).isTrue(),
+          assertThat(stream.noneMatch(predicate)).isFalse());
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    AbstractAssert<?, ?> after(Stream<T> stream, Predicate<? super T> predicate) {
+      return assertThat(stream).anyMatch(predicate);
+    }
```

Comment:
I'll split this one out to clarify that in one case the result type is changed. (Later on we'll want to automate this kind of analysis and annotate rules accordingly, so that we can warn about this on the website, and perhaps introduce a "conservative" mode.)